core/node/node_sync/src/validate_chain_ids_task.rs

```diff
@@ -32,23 +32,23 @@ impl ValidateChainIdsTask {
         gateway_chain_id: Option<SLChainId>,
         l1_client: Box<DynClient<L1>>,
         main_node_client: Box<DynClient<L2>>,
-        gateway_client: Option<Box<DynClient<L1>>>,
+        gateway_client: Option<Box<dyn EthInterface>>,
     ) -> Self {
         Self {
             l1_chain_id,
             l2_chain_id,
             gateway_chain_id,
             l1_client: l1_client.for_component("chain_ids_validation"),
             main_node_client: main_node_client.for_component("chain_ids_validation"),
-            gateway_client: gateway_client.map(|c| c.for_component("chain_ids_validation")),
+            gateway_client,
         }
     }
 
     async fn check_client(
-        l1_client: Option<Box<DynClient<L1>>>,
+        gateway_client: Option<Box<dyn EthInterface>>,
         expected: Option<SLChainId>,
     ) -> anyhow::Result<()> {
-        let (Some(l1_client), Some(expected)) = (l1_client, expected) else {
+        let (Some(l1_client), Some(expected)) = (gateway_client, expected) else {
```

Comment:
Bikeshedding: The old arg / variable naming was outdated already, but I don't understand the new one either. I'd name both the arg and the binding simply `client` given that `check_client()` is called both with L1 client and gateway clients.